### PR TITLE
Add zlib

### PR DESCRIPTION
**Pull request by ocefpaf**
Built on top of #334. @groutr are you OK with this PR? I figure you are busy. I wante to get `hdf4` and `libnetcdf` going, but we need to get `zlib` in the channel first.

(Closes #334)

### CODE REVIEW

**Comment by jakirkham** [comment]
LGTM. Ready for a merge?

File: recipes/zlib/bld.bat [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+cmake -G "NMake Makefiles" ^
+      -D CMAKE_BUILD_TYPE=Release ^
+      -D CMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
+      - DCMAKE_INSTALL_PREFIX:%LIBRARY_PREFIX% %SRC_DIR% ^
+      %SRC_DIR%
+if errorlevel 1 exit 1
+
+:: Build step.
+cmake --build . --target INSTALL --config Release
+if errorlevel 1 exit 1
```
Comment (jakirkham) [inline recipes/zlib/bld.bat]: We could do tests on Windows. There one calls `ctest`.

**Comment by jakirkham** [comment]
Thinking about tests on Windows. See comment above.

File: recipes/zlib/bld.bat [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - DCMAKE_INSTALL_PREFIX:%LIBRARY_PREFIX% %SRC_DIR% ^
+      %SRC_DIR%
+if errorlevel 1 exit 1
+
+:: Build step.
+cmake --build %SRC_DIR% --target INSTALL --config Release
+if errorlevel 1 exit 1
+
+:: cmake Test.
+ctest %SRC_DIR%
```
Comment (ocefpaf) [inline recipes/zlib/bld.bat]: @jakirkham I am not familiar with Windows nor `cmake`, is this OK?

Comment (ocefpaf) [inline recipes/zlib/bld.bat]: I guess it worked: https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.1128/job/ms408vpjjyo3nejj#L305

Comment (jakirkham) [inline recipes/zlib/bld.bat]: I'm still learning too. :wink:

Shouldn't need the directory. Also, add `if errorlevel 1 exit 1` on the next line.

File: recipes/zlib/bld.bat [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      %SRC_DIR%
+if errorlevel 1 exit 1
+
+:: Build step.
+cmake --build %SRC_DIR% --target INSTALL --config Release
+if errorlevel 1 exit 1
+
+:: cmake Test.
+ctest
+if errorlevel 1 exit 1
```
Comment (ocefpaf) [inline recipes/zlib/bld.bat]: > Shouldn't need the directory. Also, add `if errorlevel 1 exit 1` on the next line.

Removed the directory and added the exit error. (It is really annoying that we need to tell Windows they he is wrong when exiting.)

Comment (msarahan) [inline recipes/zlib/bld.bat]: Yeah, but even Linus shell scripts need an option for this.  I think it is 

    set -e

?  I don't know if windows has an equivalent.  Basically this practice prevents you from ever uploading half-baked packages.

Comment (jakirkham) [inline recipes/zlib/bld.bat]: Yes, it is. Would be nice if there was a `set -e` strategy for Windows. If there is one, I don't know it. Anyways, LGTM and AppVeyor is running the test available.

**Comment by jakirkham** [comment]
Thanks @ocefpaf and @groutr.

File: recipes/zlib/bld.bat (hunk 3 above)
Comment (jakirkham) [inline recipes/zlib/bld.bat]: LOL It didn't show your comment until after I posted that @msarahan.

**Pull request merged**